### PR TITLE
(CDAP-7761) Added a “cluster.name” setting.

### PR DESCRIPTION
**Pull request by chtyim**
- Support scope resolution.
- Added a method in RuntimeContext to get the value.

Fix for https://issues.cask.co/browse/CDAP-7761

### CODE REVIEW

**Comment by chtyim** [comment]
Building http://builds.cask.co/browse/CDAP-DUT5184-1

**Comment by bdmogal** [comment]
@chtyim, in 3.5, we had added an ``instance.name`` property to CDAP for authorization purposes. Can we re-use/replace that property here?

**Comment by chtyim** [comment]
@bdmogal The `instance.name` is used for auth purpose, meaning it actually will be the same across different clusters (e.g. if two clusters are replicating to each other).

File: cdap-api/src/main/java/co/cask/cdap/api/RuntimeContext.java [hunk 1]
```diff
@@ -37,12 +37,17 @@
   Map<String, String> getRuntimeArguments();
 
   /**
-   * @return The application namespace
+   * @return The cluster name.
```
Comment (awholegunch) [inline cdap-api/src/main/java/co/cask/cdap/api/RuntimeContext.java]: No trailing period on Javadoc return statements, please. They are only used if the @return has another sentence that follows it.

File: cdap-api/src/main/java/co/cask/cdap/api/RuntimeContext.java [hunk 2]
```diff
@@ -37,12 +37,17 @@
...
+   */
+  String getClusterName();
+
+  /**
+   * @return The application namespace.
```
Comment (awholegunch) [inline cdap-api/src/main/java/co/cask/cdap/api/RuntimeContext.java]: ibid.

**Comment by awholegunch** [comment]
@chtyim I have edited the cdap-defaults.xml file. Please check my changes carefully.

File: cdap-api/src/main/java/co/cask/cdap/api/RuntimeContext.java (hunk 1 above)
Comment (chtyim) [reply cdap-api/src/main/java/co/cask/cdap/api/RuntimeContext.java]: Ok. I was adding it as some of the @return statement has it (https://github.com/caskdata/cdap/pull/7315/files#diff-02980aab78a135d0df7c0164de33152bR35)

**Comment by anwar6953** [comment]
Why encode the cluster name in the program options?
Wouldn't it be available from the CConfiguration option that's available in AbstractContext?

**Comment by chtyim** [comment]
@anwar6953 That's the abstraction. When a program executes, everything related to the program runtime should be in the program and program option. It doesn't really care about where does those information comes from.

**Comment by anwar6953** [comment]
LGTM

**Pull request merged**